File: 2nd_quarter/week05/whaeun/햄버거 만들기.cpp
#include <string>
#include <vector>
#include <stack>

using namespace std;

int arr[4] = {1, 2, 3, 1};


int solution(vector<int> ingredient) {
    int answer = 0;
    string str = "";
    
    for(int i = 0; i < ingredient.size(); i++){
        str += '0' + ingredient[i];
    }
    
    int pos = 0;
    
    while (true) {
        int position = str.find("1231", pos);
        
        if (position == -1){
            break;
        }
        
        str.erase(position, 4);
        
        if (position > 2){
            pos = position - 3;
        }
            
        answer++;
    }
    
    return answer;
}

File: 2nd_quarter/week05/pepe/햄버거만들기.py
# 스택?

def solution(ingredient):
    answer = 0

    check = []

    for i in ingredient:
        check.append(i)
        if check[-4:] == [1, 2, 3, 1]:
            answer += 1
            for _ in range(4):
                check.pop()

    return answer


ingredient = [2, 1, 1, 2, 3, 1, 2, 3, 1]	
print(solution(ingredient))
